lighthouse-core/report/html/renderer/dom.js

```diff
@@ -27,14 +29,14 @@ class DOM {
     this._document = document;
   }
 
-  // TODO(bckenny): can pass along `createElement`'s inferred type
   /**
-   * @param {string} name
+   * @template {string} T
+   * @param {T} name
```

Comment:
can we type this as the keys of the `HTMLElementTagNameMap`?

Reply:
> can we type this as the keys of the `HTMLElementTagNameMap`?

We could, though it would eliminate the fallback to the generic `HTMLElement` when a tag name isn't recognized. Maybe we're ok with that in exchange for better completion/stricter checking?

Reply:
would we still get helpful completion if it were strict options | string? :)

Reply:
> strict options | string

sadly no, it's smart about it and coalesces to just `string`.

Reply:
ok, I tried to use just `HTMLElementTagNameMap`, and we have one place that's unrecognized: `'summary'` in

https://github.com/GoogleChrome/lighthouse/blob/a203a36026c722895d3c885e282adaf87225f19e/lighthouse-core/report/html/renderer/category-renderer.js#L191

When Edge gets `<details>` support I imagine that they'll also support `<summary>` and their WebIDL will get updated and so the tsc types will get updated (I'm surprised it already has `'details'` support, honestly), but until then we probably just want to go with this as this PR is now.